pkg/labels/labels.go

```diff
@@ -248,6 +248,17 @@ func (l *Label) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// GetExtendedKey returns the key of a label with the source encoded.
+func (l *Label) GetExtendedKey() string {
+	return common.BaseLabelSourceExtPrefix + l.Source + common.PathDelimiter + l.Key
```

Comment:
Why is `common.BaseLabelSourceExtPrefix` needed? Every label in the EndpointSelector will always have a prefix attached. Can't we just always prefix the key with just `l.Source + common.PathDelimiter`?